pkg/types/vsphere/defaults/platform.go

```diff
@@ -1,11 +1,25 @@
 package defaults
 
 import (
+	"fmt"
+
 	"github.com/openshift/installer/pkg/types"
 	"github.com/openshift/installer/pkg/types/vsphere"
 )
 
 // SetPlatformDefaults sets the defaults for the platform.
 func SetPlatformDefaults(p *vsphere.Platform, installConfig *types.InstallConfig) {
-
+	// We need to deploy templates (OVA) via DeploymentZones
+	// since we could have compute (workers) in those zones
+	// but _not_ control plane nodes. If the placementConstraints
+	// are not defined we must use the default for the datacenter
+	// and cluster.
+	// todo: jcallen: does this really belong here?
```

Comment:
I _think_ this can be removed since we are performing this same operation within machine: https://github.com/openshift/installer/pull/6770/commits/d56664c64ae8a09ee7dbbc5875e8aecafbb579b6#diff-a13f70d3a1cdbc178348f1d69e11aefee254d13faac647543cb7e4c3d73e51f0R99

Reply:
are we reflecting the calculated resource pool back to the failure domain within the machine?

Reply:
Thanks @rvanderp3 I didn't think of infra spec and terraform. We still need this. I will remove the comments.